### ISSUE TITLE
Events Module 7.0.1 Error Log

### ISSUE THREAD

**Issue by PeterCasper** [DNNCommunity/DNN.Events#92]
G'day All,

Testing the 7.0.1 release & found a couple of problems. Not a disaster yet because I should be able to roll back.

I have multiple sub-calendars modules. These are currently not displayed on my accumulator calendar. No log entry was generated. I tried to re-add them but they are not listed as available in the Sub-Calendar tab. I presume there are faults in the sub-calendars that render them ineligible to be added.

The default module view on the sub-calendars is "List" I tried displaying a sub-calendar in monthly view & got the attached error log.

[errorlog.zip](https://github.com/DNNCommunity/DNN.Events/files/2148019/errorlog.zip)

Peter C

**Comment by PeterCasper** [comment]
Time for me to develop a new plan. My IIS rolled back OK but not my database & there is a data type conversion error when using the old module with the new database structure.

Perhaps I should update my CV.

**Comment by EPTamminga** [comment]
Oops.
We use the Evotiva backup module all our production sites, that works perfectly...

**Comment by PeterCasper** [comment]
Ernst,

there are too many odd events for this to be simply a bug. I must presume that the problem is at my end. I have been using this module for several years in manners you probably never expected hence I must suspect that the problem is here, possibly in the data or alternately in my configuration.

A second portal used the Events module in 2014, then deleted the module without deleting the data. I re-added the module & removed its events, locations & categories then removed the module. Events is now crashing less frequently but there are odd results when I attempt to change views including cancelled events reappearing & recurring events missing.

There is now data in several tables, all from one portal:

EventsMaster is empty.
EventsNotification is empty.
EventsPPErrorLog is empty.
EventsSignups is empty.
EventsSubscription is empty.

EventLog holds **998,632** records including many "SCHEDULER_EVENT_FAILURE" records.
EventLogConfig holds 125 records that look ordinary
EventLogTypes holds 152 records that look ordinary
EventQueue holds 61 records. The most recent has ProcessorType misaligned & an odd looking Attributes entry.
Events holds 716 records. I have not yet detected any data problems.
EventsCategory holds 5 records & looks OK.
EventsLocation holds 3 records & looks OK.
EventsRecurMaster holds 310 records.

EventsRecurMaster$ has no key & is empty.
Events$ has no key & is empty.
EventsCategory$ has no key & holds similar but not identical data to EventsCategory.
EventsLocation$ has no key & holds similar but not identical data to EventsLocation.

- Should the last 4 tables be deleted?
- When a recurring event is created should all the events then be created in the Events table or only when an exception is generated?
- Where do I find the Events module configuration?
- May I delete EventLog records older than 90 days?

Thank you.
Peter C

**Comment by EPTamminga** [comment]
Hi Peter,

A mix of issues.

## Eventlog etc. tables

Event* (without the *s*) are tables of DNN itself, not from the Events module. Sebastian Leupold has created SQL queries to do a cleanup/trim of these tables.

## Event*$ tables

I assume they are temp tables, maybe the result of some copy action in the database. Not familiar to me

## Recurring events

At some point in the releases, we changed recurring events from a calculation only mode into records in the events table that are related by a master recurrence. Since you are working with a recent Events module, recurring events are stored as master definition record & generated individual events for every occurrence in the cycle.

## Events Module configuration

Configuration data is stored in the DNN Standard settings tables (ModuleSettings, PotalSettings and HostSettings)

**Comment by PeterCasper** [comment]
Ernst,

Thank you. I have now cleaned up the tables to the best of my ability.

I created a new test system on a clean database. I messed about with language (en-au) & timezone (Brisbane) & added events then started with 2 new events in the previous month. I adjusted the list range to -60 to +730 &  get errors, particularly when trying to swap from list view to monthly.

Peter C
[errorlog.zip](https://github.com/DNNCommunity/DNN.Events/files/2169054/errorlog.zip)
